### ISSUE TITLE
Importing new @angular/platform-browser/animations bundle

### ISSUE THREAD

**Issue by fknop** [fuse-box/fuse-box#324]
Hello,

Angular recently changed their animation bundles, it now comes inside another npm package (which is `platform-browser`. 

Here is the directory structure of platform-browser:

```
package.json (links to bundles/platform-browser.umd.js)
bundles/ (directory of bundles)
animations/
   package.json (links to ../bundles/platform-browser-animations.umd.js)
```

The problem is that it looks like fuse-box does not resolve the animations bundle with the package.json of the animations folder.

It only resolves the main bundle from the main package.json, so the code is not imported correctly and I have a 404 during runtime.

Any idea how I could solve this ?

**Comment by nchanged** [comment]
Hi! yes, we have this problem. It's a known bug.

Would you consider using [aliases](http://fuse-box.org/#alias) for now?  Or you could reference it directly. There is a fix in mind, stay tuned. ;-)

**Comment by nchanged** [comment]
Something like that could work:
`require("@angular/platform-browser/platform-browser-animations.umd")`

**Comment by fknop** [comment]
Thanks, sorry I didn't find the existing issue !

**Comment by fknop** [comment]
Aliases are working, thank you.

**Comment by PinkTiu** [comment]
@fknop How did you manage that?

First I tried to use this alias to import the animations module: 
```
alias: {
    '@angular/platform-browser/animations': '@angular/platform-browser/bundles/platform-browser-animations.umd.js',
},
```


Then I got this error
```
@angular/platform-browser/animations not found on request
n.onreadystatechange @ undefined@undefined:7
wrapFn @ undefined@undefined:7
ZoneDelegate.invokeTask @ undefined@undefined:7
Zone.runTask @ undefined@undefined:7
ZoneTask.invoke @ undefined@undefined:7
```


So I imported the `@angular/platform-browser/animations` like this: 
```
alias: {
    '@angular/platform-browser/animations': '@angular/platform-browser/bundles/platform-browser-animations.umd.js',
    '@angular/animations/browser': '@angular/animations/bundles/animations.umd.js',
},
```

This leads to another error:
```
undefined@undefined:7 Uncaught TypeError: Cannot read property 'prototype' of undefined
    at __extends (undefined@undefined:7)
    at undefined@undefined:7
    at undefined@undefined:7
    at undefined@undefined:7
    at Number.<anonymous> (undefined@undefined:7)
    at c (undefined@undefined:7)
    at w.require (undefined@undefined:7)
    at Number.<anonymous> (app.module.ts:25)
    at c (undefined@undefined:7)
    at w.require (undefined@undefined:7)
```

Do you have a hint? How did you managed it?

**Comment by nchanged** [comment]
@PinkTiu these guys are changing the API every month, so annoying. I had to lock dependencies in the project...

**Comment by kylecordes** [comment]
This "just works" with NPM module resolution. If fuse-box could be tweaked to follow work with module resolution normally, then this library and others would work fine.

**Comment by nchanged** [comment]
This version should fix the bug `fuse-box@2.1.0-beta.7` please verify

**Comment by Tenvan** [comment]
did not work, gives this error:

Connecting to fusebox HMR at ws://localhost:4445
vendor.js:23274 Uncaught TypeError: Cannot read property 'ɵDomAnimationEngine' of undefined
    at vendor.js:23274
    at vendor.js:23074
    at Number.<anonymous> (vendor.js:23080)
    at c (vendor.js:99090)
    at w.require (vendor.js:99090)
    at Number.<anonymous> (app.module.ts:2)
    at c (vendor.js:99090)
    at w.require (vendor.js:99090)
    at Number.<anonymous> (main.ts:5)
    at c (vendor.js:99090)
(anonymous) @ vendor.js:23274
(anonymous) @ vendor.js:23074
(anonymous) @ vendor.js:23080
c @ vendor.js:99090
w.require @ vendor.js:99090
(anonymous) @ app.module.ts:2
c @ vendor.js:99090
w.require @ vendor.js:99090
(anonymous) @ main.ts:5
c @ vendor.js:99090
r.import @ vendor.js:99090
(anonymous) @ todo.service.ts:5
(anonymous) @ todo.service.ts:5
vendor.js:99090 @angular/platform-browser/bundles/platform-browser-animations.umd.js/browser not found on request
i.onreadystatechange @ vendor.js:99090
wrapFn @ vendor.js:3174
ZoneDelegate.invokeTask @ vendor.js:2543
Zone.runTask @ vendor.js:2310
ZoneTask.invoke @ vendor.js:2605
vendor.js:157 Connected

**Comment by nchanged** [comment]
@Tenvan i personally tested it and it works. Make sure you have cleared the cache. I can provide a test repository

**Comment by nchanged** [comment]
@Tenvan unless there is something fishy. Could you share a github repo with the problem above?

**Comment by Tenvan** [comment]
yes, of course:
https://github.com/Tenvan/angular2-example/tree/feature/ng4-upgrade

**Comment by nchanged** [comment]
@Tenvan interestingly enough, it launched without any problems. 

```
└── default (0 files,  53 Bytes)
└── fusebox-hot-reload (1 files) 1.8 kB
└── fusebox-websocket (1 files) 2.9 kB
└── events (1 files) 9.8 kB
└── core-js (81 files) 57.1 kB
└── zone.js (1 files) 65.7 kB
└── process (1 files) 3 kB
└── object-assign-polyfill (1 files) 1 kB
└── @angular/platform-browser-dynamic (1 files) 7.1 kB
└── @angular/compiler (1 files) 1 MB
└── @angular/core (1 files) 469.5 kB
└── rxjs (70 files) 142.8 kB
└── @angular/platform-browser (1 files) 156.2 kB
└── @angular/common (1 files) 127.5 kB
└── @angular/forms (1 files) 221.6 kB
└── @angular/material (1 files) 739.6 kB
└── @angular/http (1 files) 78.1 kB
└── http (1 files) 288 Bytes
└── lodash (1 files) 527.2 kB
```

Like a charm ;-)  Maybe you do a fresh clone on your own repo?

**Comment by nchanged** [comment]
Unless your forgot to upgrade dependencies.

**Comment by Tenvan** [comment]
my output iss different:

└── default (0 files,  53 Bytes)
└── fusebox-hot-reload (1 files) 1.8 kB
└── fusebox-websocket (1 files) 2.9 kB
└── events (1 files) 9.8 kB
└── core-js (81 files) 57.1 kB
└── zone.js (1 files) 92.2 kB
└── @angular/platform-browser-dynamic (1 files) 6.7 kB
└── @angular/compiler (1 files) 973.1 kB
└── @angular/core (1 files) 460.5 kB
└── rxjs (75 files) 160.7 kB
└── @angular/common (1 files) 129 kB
└── @angular/platform-browser (3 files) 152.7 kB
└── @angular/forms (1 files) 202.4 kB
└── @angular/material (1 files) 670.5 kB
└── @angular/http (1 files) 74 kB
└── http (1 files) 288 Bytes
└── lodash (1 files) 527.1 kB

this two outputs are missing:
└── process (1 files) 3 kB
└── object-assign-polyfill (1 files) 1 kB

did you install with yarn or npm ?

**Comment by nchanged** [comment]
I installed it with npm. Yarn tends to lock deps, maybe you remove yarn.lock?

**Comment by Tenvan** [comment]
nope.. same error with clean npm-installation:

```
Connecting to fusebox HMR at ws://localhost:4445
vendor.js:58550 Uncaught TypeError: Cannot read property 'ɵDomAnimationEngine' of undefined
    at vendor.js:58550
    at vendor.js:58350
    at Number.<anonymous> (vendor.js:58356)
    at c (vendor.js:99090)
    at w.require (vendor.js:99090)
    at __extends.__extends.Object.setPrototypeOf.__proto__ (vendor.js:64706)
    at Number.<anonymous> (vendor.js:64734)
    at c (vendor.js:99090)
    at w.require (vendor.js:99090)
    at Number.<anonymous> (app.module.ts:4)
(anonymous) @ vendor.js:58550
(anonymous) @ vendor.js:58350
(anonymous) @ vendor.js:58356
c @ vendor.js:99090
w.require @ vendor.js:99090
__extends.__extends.Object.setPrototypeOf.__proto__ @ vendor.js:64706
(anonymous) @ vendor.js:64734
c @ vendor.js:99090
w.require @ vendor.js:99090
(anonymous) @ app.module.ts:4
c @ vendor.js:99090
w.require @ vendor.js:99090
(anonymous) @ main.ts:7
c @ vendor.js:99090
r.import @ vendor.js:99090
(anonymous) @ todo-item.component.ts:8
(anonymous) @ todo-item.component.ts:8
vendor.js:99090 @angular/platform-browser/bundles/platform-browser-animations.umd.js/browser not found on request
```

which node.js version do you use?

```
{ 'fusebox-angular': '1.0.0',
  npm: '4.5.0',
  ares: '1.10.1-DEV',
  cldr: '30.0.3',
  http_parser: '2.7.0',
  icu: '58.2',
  modules: '51',
  node: '7.8.0',
  openssl: '1.0.2k',
  tz: '2016j',
  unicode: '9.0',
  uv: '1.11.0',
  v8: '5.5.372.43',
  zlib: '1.2.11' }
```

**Comment by Tenvan** [comment]
installed packages after npm install:

```
+-- @angular/common@4.1.3                                             
+-- @angular/compiler@4.1.3                                           
+-- @angular/core@4.1.3                                               
+-- @angular/forms@4.1.3                                              
+-- @angular/http@4.1.3                                               
+-- @angular/material@2.0.0-beta.5                                    
+-- @angular/platform-browser@4.1.3                                   
+-- @angular/platform-browser-dynamic@4.1.3                           
+-- @angular/router@4.1.3                                             
+-- @types/chai@3.5.2                                                 
+-- @types/core-js@0.9.41                                             
+-- @types/lodash@4.14.64                                             
+-- @types/mocha@2.2.41                                               
+-- @types/node@7.0.21                                                
+-- @types/sinon@2.2.2                                                
+-- angular-in-memory-web-api@0.3.2                                   
+-- chai@3.5.0                                                        
+-- core-js@2.4.1                                                     
+-- cpx@1.5.0                                                         
+-- cross-env@5.0.0                                                   
+-- fuse-box@2.1.0-beta.7                                             
+-- jsdom@10.1.0                                                      
+-- jsdom-no-contextify@3.1.0                                         
+-- lodash@4.17.4                                                     
+-- mocha@3.4.1                                                       
+-- node-sass@4.5.3                                                   
+-- nyc@10.3.2                                                        
+-- rimraf@2.6.1                                                      
+-- rxjs@5.4.0                                                        
+-- sinon@2.3.0                                                       
+-- ts-node@3.0.4                                                     
+-- typescript@2.3.2                                                  
`-- zone.js@0.8.11                                                    
```

**Comment by nchanged** [comment]
`v7.8.0`
It's strange. I had no problems whatsoever. Cloned, npm install, hit the browser.

**Comment by nchanged** [comment]
@Tenvan i see, you didn't commit package.json
```
"dependencies": {
    "@angular/common": "2.3.1",
    "@angular/compiler": "2.3.1",
    "@angular/core": "2.3.1",
    "@angular/forms": "2.3.1",
    "@angular/http": "2.3.1",
    "@angular/material": "2.0.0-beta.2",
    "@angular/platform-browser": "2.3.1",
    "@angular/platform-browser-dynamic": "2.3.1",
    "@angular/router": "3.3.1",
    "angular-in-memory-web-api": "0.2.1",
    "core-js": "2.4.1",
    "lodash": "4.17.3",
    "rxjs": "5.0.1",
    "zone.js": "0.7.2"
  }
```
Could you save it all there?

**Comment by Tenvan** [comment]
you have to checkout the correct branch ^^
my posted links directs to them: feature/ng4-upgrade

**Comment by nchanged** [comment]
sorry, my bad. Checking it now

**Comment by nchanged** [comment]
I see, i guess we need to favor `module` over `main` property in `package.json` if present

**Comment by nchanged** [comment]
One thing i don't get, what is this ` require('@angular/platform-browser/bundles/platform-browser-animations.umd.js/browser')`

Doesn't seem to be a valid package

**Comment by nchanged** [comment]
@Tenvan ok i found the problem
first:
```
npm install @angular/animations
```

second:

remove your alias config, it has been fixed in beta.7

But now there is a different problem with angular-materials, I hope not fuse-box related

**Comment by Tenvan** [comment]
yepp.. this themes to be an 'angular4 animation' problem... but i cant figure out, how to resolve this :/

**Comment by nchanged** [comment]
read above ^ =)

**Comment by Tenvan** [comment]
ok.. i try..

**Comment by Tenvan** [comment]
ahhh.. one step further :)
ich will investigate material-problem ...

**Comment by nchanged** [comment]
Cool thanks!! at least it resolves it correctly.

**Comment by Tenvan** [comment]
ok.. now all works, also material...
i have made an pull-request, so you can merge the changes.

**Comment by nchanged** [comment]
Oh!! so cool man! Thanks a lot, i will merge it at home along with some tweaks (WebIndexPlugin)

**Comment by nchanged** [comment]
Fixed in `2.1.0-beta.4`
